model/src/main/java/org/eclipse/rdf4j/model/impl/GraphImpl.java

```diff
@@ -1,173 +0,0 @@
-/*******************************************************************************
- * Copyright (c) 2015 Eclipse RDF4J contributors, Aduna, and others.
- * All rights reserved. This program and the accompanying materials
- * are made available under the terms of the Eclipse Distribution License v1.0
- * which accompanies this distribution, and is available at
- * http://www.eclipse.org/org/documents/edl-v10.php.
- *******************************************************************************/
-package org.eclipse.rdf4j.model.impl;
-
-import java.io.IOException;
-import java.io.ObjectInputStream;
-import java.io.ObjectOutputStream;
-import java.util.AbstractCollection;
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.LinkedList;
-
-import org.eclipse.rdf4j.OpenRDFUtil;
-import org.eclipse.rdf4j.model.Graph;
-import org.eclipse.rdf4j.model.IRI;
-import org.eclipse.rdf4j.model.Resource;
-import org.eclipse.rdf4j.model.Statement;
-import org.eclipse.rdf4j.model.Value;
-import org.eclipse.rdf4j.model.ValueFactory;
-import org.eclipse.rdf4j.util.iterators.FilterIterator;
-
-/**
- * Basic implementation of Graph.
- * 
- * @deprecated since release 2.7.0. Use a {@link org.eclipse.rdf4j.model.Model} implementation (e.g. {@link TreeModel}
- *             or {@link LinkedHashModel} instead.
- * @author Arjohn Kampman
- */
-@Deprecated
-public class GraphImpl extends AbstractCollection<Statement> implements Graph {
-
-	private static final long serialVersionUID = -5307095904382050478L;
-
-	protected LinkedList<Statement> statements;
```

Comment:
GraphImpl was using a LinkedList, which is usually slow compared to an ArrayList for inserts until the List grows very large (gigabytes). 

The Model implementations use either a TreeSet or a HashSet approach. I think both of these are slower than both LinkedList and ArrayList for workloads with mostly inserts. 

How does the workload compare for the places where you are now switching out GraphImpl for one of the Model implementations?

Reply:
No idea, I haven't done any performance tests on this. 

It wouldn't be hard to introduce a new list-based Model implementation to facilitate quicker inserts, if there's a need for that.